optimum/habana/transformers/generation/utils.py

```diff
@@ -2952,7 +2953,8 @@ def expand_if_needed(tensor, new_size, value, dim=-1):
                 if self.generation_config.early_stopping:
                     num_eos_tokens.add_(beam_tokens[0:num_beams].eq(self.config.eos_token_id).sum())
 
-                beam_scores.add_(torch.where(beam_tokens.eq(self.config.eos_token_id), float("-inf"), 0.0))
+                if self.config.eos_token_id:
```

Comment:
```suggestion
                if self.config.eos_token_id is not None:
```

Is this your intention for this conditional?